### PR TITLE
feat(openclaw-telemetry-cli): one-shot installer in a separate npm package (0.0.7)

### PR DESCRIPTION
**Pull request by csansoon**
## Summary

Brings back the `npx -y @latitude-data/openclaw-telemetry-cli install` one-shot UX that PR #2920 removed when it deleted the bundled CLI from the runtime package. Living in a **separate** npm package means this CLI is installed via `npx` / `npm install -g` rather than `openclaw plugins install` — so it never goes through OpenClaw's install-time security scanner. The runtime side stays clean (no `child_process`, no `node:fs` runtime reads); the operator side stays a one-shot.

The CLI carries the deleted CLI's core forward (cli.ts, setup.ts, settings-file.ts, openclaw-cli.ts + tests) and adds the new behaviour from the planning conversation:

- **Configurable config dir.** `--openclaw-dir <path>` > `OPENCLAW_HOME` env > `./openclaw.json` in cwd > `~/.openclaw` default. Passed through as `OPENCLAW_HOME` to every spawned `openclaw` subprocess; the post-write `openclaw config validate --json` step catches the case where OpenClaw doesn't honor the env var (restores backup + aborts loudly with both paths in the error rather than silently writing to two places).
- **npm-spec install.** `applyChanges` spawns `openclaw plugins install @latitude-data/openclaw-telemetry@0.0.7 --force` (pinned, lockstep with the runtime) instead of installing from a local package root. Pinning also satisfies OpenClaw's `Pin install specs to exact versions` supply-chain audit warning automatically.
- **Atomic settings writes.** `writeSettings` goes through `<file>.tmp.<pid>` + `renameSync`. Combined with the `.latitude-bak` backup, mid-write crashes can no longer corrupt `openclaw.json`.
- **Post-write `openclaw config validate --json`.** Catches schema regressions and config-dir mismatch *immediately* — before the operator restarts the gateway and finds out the hard way. Restores the backup and aborts with the validator's message on failure.
- **Upgrade-detection UX.** Reads `<configDir>/plugins/installs.json` to render `Upgrading 0.0.6 → 0.0.7` (or `Re-applying 0.0.7 (idempotent)`) before any subprocess spawn.
- **`--dry-run`.** Resolves config dir, prints the exact `openclaw plugins install` command, prints a unified-ish JSON diff between the current `openclaw.json` and the proposed result. Zero subprocess calls, zero writes.
- **Gateway restart prompt.** Default: prompt on TTY (`Restart the OpenClaw gateway now to load the plugin? [Y/n]`), skip non-TTY with manual instructions. `--restart` always restarts (CI escape hatch), `--no-restart` always skips (always wins).
- **Lockstep version refusal.** CLI hardcodes `RUNTIME_VERSION = "0.0.7"` and queries the npm registry to verify that exact runtime version exists before installing. Aborts with `Upgrade the CLI: npm install -g @latitude-data/openclaw-telemetry-cli@latest` on 404. Network failure → warn and continue (the install spawn would fail visibly anyway).

A standalone publish workflow at `.github/workflows/publish-openclaw-telemetry-cli.yml` mirrors `publish-claude-code-telemetry.yml`: path filter on `packages/telemetry/openclaw-cli/**`, npm version-bump check, GitHub release with extracted CHANGELOG section, reuses the existing `NPM_TOKEN` secret.

## Lockstep policy

CLI 0.0.7 installs runtime 0.0.7 (PR #2925). Bumping the runtime requires bumping `RUNTIME_VERSION` in `src/setup.ts` and re-publishing both packages in the same release. The README documents this. The npm-registry pre-flight check enforces it at install time (the CLI refuses to install a runtime version that's not on npm).

## Test plan

- [x] `pnpm --filter @latitude-data/openclaw-telemetry-cli typecheck` — passes
- [x] `pnpm --filter @latitude-data/openclaw-telemetry-cli test` — 68 passed (carryover settings-file 13 + openclaw-cli 6 + new config-dir 7 + diff 5 + cli flag-parsing 11 + settings-file FS-touching 11 + installs.json reader 5)
- [x] `pnpm --filter @latitude-data/openclaw-telemetry-cli check` — biome clean
- [x] `pnpm --filter @latitude-data/openclaw-telemetry-cli build` — produces `dist/cli.js` with shebang + exec bit + `RUNTIME_VERSION = "0.0.7"` baked
- [ ] **After publish, fresh install:** `npx -y @latitude-data/openclaw-telemetry-cli@0.0.7 install` → prompts for apiKey + project, spawns the npm-spec install, writes `plugins.entries[id]`, adds to `plugins.allow`, runs validate, prompts to restart gateway. Expected: traces flow within seconds of the restart.
- [ ] **Re-install (upgrade UX):** running on top of an existing 0.0.7 → `Re-applying 0.0.7 (idempotent)`. Existing hand-edited `debug: true` preserved.
- [ ] **CI install:** `npx -y @latitude-data/openclaw-telemetry-cli@0.0.7 install --api-key=lat_xxx --project=my-project --yes --no-restart` → zero prompts, exits 0, gateway not restarted.
- [ ] **Dry-run:** `--dry-run --api-key=k --project=p` → prints resolved dir + install spec + JSON diff, exits 0, `openclaw.json` unchanged.
- [ ] **Custom config dir:** `OPENCLAW_HOME=/tmp/test npx -y @latitude-data/openclaw-telemetry-cli@0.0.7 install --yes --api-key=k --project=p` → writes to `/tmp/test/openclaw.json`; on validate failure (OpenClaw didn't honor the env), restores backup and prints both paths in the error.
- [ ] **Uninstall:** `npx -y @latitude-data/openclaw-telemetry-cli@0.0.7 uninstall --yes` → spawns `openclaw plugins uninstall ... --force`, removes entry, removes from `plugins.allow`, sweeps legacy env keys, gateway-restart prompt per `--yes`/`--no-restart`.
- [ ] **Audit clean:** `openclaw security audit --deep` → zero warnings about `@latitude-data/openclaw-telemetry-cli` (was never `openclaw plugins install`'d).

### CODE REVIEW

File: packages/telemetry/openclaw-cli/src/setup.ts [hunk 1]
```diff
@@ -0,0 +1,678 @@
...
+} {
+  const [subcommand, ...rest] = argv
+  const flags: Record<string, string | boolean> = {}
+  for (const arg of rest) {
+    if (!arg.startsWith("--")) continue
+    const eq = arg.indexOf("=")
+    if (eq >= 0) {
+      flags[arg.slice(2, eq)] = arg.slice(eq + 1)
+    } else {
+      flags[arg.slice(2)] = true
```
[the remark this answers is not in the document]
Comment (csansoon) [reply packages/telemetry/openclaw-cli/src/setup.ts]: Fixed in 9c3b2bf8c. Extended `parseFlags` to accept the space-separated form (`--key value`) for the value-taking flags (`--api-key`, `--project`, `--openclaw-dir`). Boolean flags can't accidentally swallow adjacent flags as values — `--no-content --yes` still parses as two booleans because the next-token-is-a-flag guard short-circuits. Added 4 regression tests covering the new form, end-of-argv, and the `--key --otherflag` edge case.

File: packages/telemetry/openclaw-cli/src/setup.ts [hunk 2]
```diff
@@ -0,0 +1,678 @@
...
+  const plan = [
+    `Run \`openclaw plugins uninstall ${PLUGIN_ID} --force\` (removes files, install record, and plugin entry)`,
+    `Sweep any leftover LATITUDE_* keys from settings.env`,
+    `Backup of openclaw.json saved at ${paths.settingsBackupPath}`,
+  ]
+  note(plan.join("\n"), "Plan")
+
+  if (!flags.noPrompt && process.stdin.isTTY === true) {
+    const ok = await confirm({ message: "Proceed?", initialValue: true })
+    if (isCancel(ok) || ok !== true) return onCancel()
```
[the remark this answers is not in the document]
Comment (csansoon) [reply packages/telemetry/openclaw-cli/src/setup.ts]: Fixed in 9c3b2bf8c. Non-TTY uninstall now throws `Non-interactive uninstall requires --yes / --no-prompt to confirm.` rather than silently going ahead. Symmetric with the install side that throws when `--api-key`/`--project` aren't provided non-interactively.

File: packages/telemetry/openclaw-cli/src/setup.ts [hunk 3]
```diff
@@ -0,0 +1,678 @@
...
+    const detail =
+      validateResult.ok === false
+        ? validateResult.stderr.trim() || validateResult.stdout.trim() || `exit code ${validateResult.code}`
+        : validateResult.stdout.trim() || "config validate reported invalid config"
+    throw new Error(
+      `openclaw config validate reported a problem after our changes:\n  ${detail}\n` +
+        (restored
+          ? `Backup restored from ${paths.settingsBackupPath} — your config is back to the pre-install state.`
+          : `Backup at ${paths.settingsBackupPath} could not be restored automatically; restore it manually.`),
+    )
```
[the remark this answers is not in the document]
Comment (csansoon) [reply packages/telemetry/openclaw-cli/src/setup.ts]: Fixed in 9c3b2bf8c. Two-tier rollback: prefer `.latitude-bak` if it exists, else write a snapshot we capture right after `openclaw plugins install` succeeded but before our config layer goes on. The snapshot is always schema-valid (it's just OpenClaw's bookkeeping with a disabled plugin entry), so a fresh-install validate failure now leaves a clean rollback state. Recovery message also points at `openclaw plugins uninstall` for full revert in case the operator wants to start over.

File: packages/telemetry/openclaw-cli/src/setup.ts [hunk 4]
```diff
@@ -0,0 +1,678 @@
...
+  try {
+    const res = await fetch(url, { method: "GET", signal: AbortSignal.timeout(5_000) })
+    if (res.status === 404) {
+      cancel(
+        `CLI ${process.env.npm_package_version ?? "(this version)"} expects ` +
+          `${RUNTIME_PACKAGE_NAME}@${RUNTIME_VERSION} but npm doesn't have that exact version. ` +
+          `Upgrade the CLI: npm install -g @latitude-data/openclaw-telemetry-cli@latest`,
+      )
+      process.exit(1)
+    }
```
[the remark this answers is not in the document]
Comment (csansoon) [reply packages/telemetry/openclaw-cli/src/setup.ts:579]: Fixed in 9c3b2bf8c. Extracted `readCliVersion` into `src/version.ts` as a shared helper between `cli.ts` (already used the package.json read for `--version`) and `setup.ts` (now uses it in the npm-404 abort message). `process.env.npm_package_version` removed — agreed it's unset for npx/global.

File: packages/telemetry/openclaw-cli/README.md [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+  --no-restart
+```
+
+`--no-restart` keeps the gateway running while you finish other config; restart yourself when ready.
+
+## Config dir resolution
+
+The CLI resolves the OpenClaw config directory in this order, stopping at the first match:
+
+1. **`--openclaw-dir <path>`** flag (absolute, or relative to cwd).
```
[the remark this answers is not in the document]
Comment (csansoon) [reply packages/telemetry/openclaw-cli/README.md]: Fixed in 9c3b2bf8c. The README's config-dir resolution section now uses `--openclaw-dir=<path>` (\=-form) with a parenthetical noting the space-separated form also works — same root issue as the parser comment, and the parser fix in this commit means both forms are valid going forward. Operators shouldn't have to grep the parser to figure out which syntax is supported.

**Review by andresgutgon — APPROVED** [review]
